OutOfSchool/OutOfSchool.WebApi/Services/ApplicationService.cs

```diff
@@ -207,21 +208,28 @@ public async Task<SearchResult<ApplicationDto>> GetAllByParent(Guid id, Applicat
             logger.LogInformation($"Getting Applications by Parent Id started. Looking Parent Id = {id}.");
             FilterNullValidation(filter);
 
-            Expression<Func<Application, bool>> applicationFilter = a => a.ParentId == id;
+            var predicate = PredicateBuild(filter);
+            predicate = predicate.And(a => a.ParentId == id);
 
-            var totalAmount = await applicationRepository.Count(where: applicationFilter).ConfigureAwait(false);
+            var sortPredicate = SortExpressionBuild(filter);
+
+            var totalAmount = await applicationRepository.Count(where: predicate).ConfigureAwait(false);
 
-            var applications = applicationRepository.Get<int>(skip: filter.From, take: filter.Size, where: applicationFilter, includeProperties: "Workshop,Child,Parent");
+            var applications = await applicationRepository.Get<Application>(
+                skip: filter.From,
+                take: filter.Size,
+                where: predicate,
+                includeProperties: "Workshop,Child,Parent",
+                orderBy: sortPredicate).ToListAsync().ConfigureAwait(false);
```

Comment:
not a call to immediate action, but as per today's discussion maybe these 3 methods you've rewrote are good canditates for `asNoTracking` flag

Reply:
I implemented it to Application, Parent and Provider controllers. Lets look at the result ))